macosx/PortChecker.mm

```diff
@@ -112,30 +66,48 @@ - (void)startProbe:(NSTimer*)timer
 {
     self.fTimer = nil;
 
-    NSString* urlString = [NSString stringWithFormat:@"https://portcheck.transmissionbt.com/%ld", [timer.userInfo integerValue]];
+    NSString* urlString = [NSString stringWithFormat:@"https://portcheck.transmissionbt.com/%ld", [(NSNumber*)timer.userInfo integerValue]];
     NSURLRequest* portProbeRequest = [NSURLRequest requestWithURL:[NSURL URLWithString:urlString]
                                                       cachePolicy:NSURLRequestReloadIgnoringLocalAndRemoteCacheData
                                                   timeoutInterval:15.0];
 
-    if ((self.fConnection = [[NSURLConnection alloc] initWithRequest:portProbeRequest delegate:self]))
-    {
-        self.fPortProbeData = [[NSMutableData alloc] init];
-    }
-    else
-    {
-        NSLog(@"Unable to get port status: failed to initiate connection");
-        [self callBackWithStatus:PORT_STATUS_ERROR];
-    }
+    _fTask = [_fSession dataTaskWithRequest:portProbeRequest
+                          completionHandler:^(NSData* _Nullable data, NSURLResponse* _Nullable response, NSError* _Nullable error) {
+                              if (error)
+                              {
+                                  NSLog(@"Unable to get port status: connection failed (%@)", error.localizedDescription);
+                                  [self callBackWithStatus:PORT_STATUS_ERROR];
+                                  return;
+                              }
+                              NSString* probeString = [[NSString alloc] initWithData:data ?: NSData.data encoding:NSUTF8StringEncoding];
```

Comment:
Obj-C has elvis operator?

Reply:
Yes, it's GNU C, Objective-C, Objective-C++, available with both GCC and clang.